src/storage/garbage_collector.cpp

```diff
@@ -13,25 +17,43 @@
 namespace terrier::storage {
 
 std::pair<uint32_t, uint32_t> GarbageCollector::PerformGarbageCollection() {
+  // Create the UndoBuffer for this GC run
+  delta_record_compaction_buffer_ = new UndoBuffer(txn_manager_->buffer_pool_);
```

Comment:
What is the point of storing this on a heap with a pointer member? Can you inline it?